app/controllers/application_controller.rb

```diff
@@ -1,8 +1,23 @@
 class ApplicationController < ActionController::Base
+  include ApplicationHelper
+  before_action :configure_permitted_parameters, if: :devise_controller?
   # Only the autenticated users are able to navigate  in the app
-  before_action :authenticate_user!
+  before_action :authenticate_user!, if: :user_signed_in?
+  before_action :authenticate_user!, except: [:index], if: !:user_signed_in?
+  helper_method :followhing_user?
 
   rescue_from CanCan::AccessDenied do |exception|
     redirect_to root_path
   end
+
+
+  protected
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: [:username, :first_name, :last_name])
+  end
+
+  def followhing_user?(user)
```

Comment:
@sansuaza is a controller a good place for this method? do you need it in multiple controllers and views? maybe a helper would be a better fit? what do you think?

Reply:
Yes, deffinitelly thats a good practice, inclusive i can reuse it in the method `can_comment?` ubicated in the application helper